hat/assets/js/apps/Iaso/domains/forms/hooks/useGetPossibleFields.ts

```diff
@@ -42,6 +57,40 @@ export const useGetPossibleFields = (formId?: number, appId?: string): Result =>
     return usePossibleFields(isFetchingForm, currentForm);
 };
 
+export const useAllPossibleFields = (
+    isFetchingForms: boolean,
+    allForms: Form[] = [],
+): AllResults => {
+    return useMemo(() => {
+        const allPossibleFields: PossibleFieldsForForm[] = [];
+        allForms.forEach(form => {
+            const possibleFields =
+                form?.possible_fields?.map(field => ({
+                    ...field,
+                    fieldKey: field.name.replace('.', ''),
+                })) || [];
+            allPossibleFields.push({
+                form_id: form.form_id,
+                name: form.name,
+                possibleFields,
+            });
+        });
+
+        return {
+            allPossibleFields,
+            isFetchingForms,
+        };
+    }, [isFetchingForms, allForms]);
+};
```

Comment:
😮  Do you really get all possible fields for all forms ? this is a huge query to the backend and can be a big payload. This is one reason why we have to preselect a form to enable querybuilder on instances page